docs/arcadia-users-group/20221017-conda/lesson.md

```diff
@@ -0,0 +1,369 @@
+
+The command prompt should now be prepended by `(mynewenv)` instead of `(base)`.
+The command prompt is a useful tool to orient yourself as to which environment you're currently using.
+
+If you install a new tool while in an environment, it only gets installed in that environment.
+```
+mamba install samtools
+```
+
+Even better, you can specify which version of samtools you would like to install:
+```
+mamba install samtools=1.9
+```
+
+We just installed a very old version. 
+We can ask mamba to automatically update our package installation to the latest version:
+```
+mamba update samtools
+```
+
+The directories of the active environment’s executable files are added to the system path (this means that you can now access them more easily). 
+
+```
+which samtools
+```
+
+You can leave an environment with this command:
+
+```
+conda deactivate
+```
+
+Now, if we run the same command, we'll get a different result:
+
+```
+which samtools
+```
+
+### Searching, installing, and removing packages
+
+To list out all the installed packages in the currently active environment, run:
+
+```
+conda list
+```
+
+To search for all the available versions of a certain package, you can use the search command. 
+For instance, to list out all the versions of samtools, run:
+
+```
+conda search samtools
+```
+
+Similarly to the conda list command, this one results in a list of the matching package names, versions, and channels:
+
+```
+Loading channels: done
+# Name                       Version           Build  Channel
+samtools                      0.1.12               0  bioconda
+samtools                      0.1.12               1  bioconda
+samtools                      0.1.12               2  bioconda
+samtools                      0.1.12      hdd8ed8b_3  bioconda
+```
+
+### Freezing an environment
+
+Keeping track of the software versions you used to run your analyses is a cornerstone of reproducible science.
+Conda facilitates this in many ways.
+One handy way is to record the packages you've installed into a text file.
+
+This will save the list of **conda-installed** software you have in a particular environment to the file `packages.txt`:
+```
+conda activate mynewenv
+conda env export > mynewenv.yml
+conda deactivate
+```
+(it will not record the software versions for software not installed by conda.)
+
+```
+conda env create -n mynewenv2 --file mynewenv.yml
+```
+will install those packages in your local environment.
+
+### Encoding environments in and installing environments from text files
+
+Freezing an environment with `conda env export` is a perfectly viable option to record your compute state for a given environment.
+This approach can be brittle to operating system because so there are other more flexible ways to achieve similar things.
+
+Instead, you can keep track of software and versions you install in a YAML file.
+Open a text file using nano, vim, VS code, or your favorite text editor and paste the following text.
+
+```
+channels:
```

Comment:
I'd add a note about: Not every package can be found on one of these channels. And some may only be available through PyPI. And that conda can support pip installations in these files as well

Reply:
oh I love this! I'll add it to the bottom of the lesson!